pkg/driver/nydus/nydus.go

```diff
@@ -433,76 +436,124 @@ func (d *Driver) getChunkDict(ctx context.Context, provider accelcontent.Provide
 }
 
 // FetchRemoteCache fetch cache manifest from remote
-func (d *Driver) FetchRemoteCache(ctx context.Context, pvd accelcontent.Provider, ref string) error {
+func (d *Driver) FetchRemoteCache(ctx context.Context, pvd accelcontent.Provider, ref string, sourceDesc *ocispec.Descriptor) (*ocispec.Descriptor, error) {
 	resolver, err := pvd.Resolver(ref)
 	if err != nil {
-		return err
+		return nil, err
 	}
-
 	rc := &containerd.RemoteContext{
-		Resolver: resolver,
+		Resolver:        resolver,
+		PlatformMatcher: d.platformMC,
 	}
 	name, desc, err := rc.Resolver.Resolve(ctx, ref)
 	if err != nil {
 		if errors.Is(err, containerdErrDefs.ErrNotFound) {
 			// Remote cache may do not exist, just return nil
-			return nil
+			return nil, nil
 		}
-		return err
+		return nil, err
 	}
 	fetcher, err := rc.Resolver.Fetcher(ctx, name)
 	if err != nil {
-		return err
+		return nil, err
 	}
 	ir, err := fetcher.Fetch(ctx, desc)
 	if err != nil {
 		if errdefs.NeedsRetryWithHTTP(err) {
 			pvd.UsePlainHTTP()
 			ir, err = fetcher.Fetch(ctx, desc)
 			if err != nil {
-				return errors.Wrap(err, "try to pull remote cache")
+				return nil, errors.Wrap(err, "try to pull remote cache")
 			}
 		} else {
-			return errors.Wrap(err, "pull remote cache")
+			return nil, errors.Wrap(err, "pull remote cache")
 		}
 	}
+	defer ir.Close()
 
-	bytes, err := io.ReadAll(ir)
+	manifestIndexBytes, err := io.ReadAll(ir)
```

Comment:
We should check the `desc.MediaType` first, the possible values:

- `images.MediaTypeDockerSchema2Manifest` / `ocispec.MediaTypeImageManifest`
- `images.MediaTypeDockerSchema2ManifestList` / `ocispec.MediaTypeImageIndex`

Reply:
ok